Docker image doesn't build

### Description

```
$ docker build -t goer -f deploy/goer/Dockerfile .
Sending build context to Docker daemon  281.6kB
Step 1/7 : FROM golang:1.16.3-alpine AS build
 ---> 270727b8fd0f
Step 2/7 : WORKDIR /tmp/goer
 ---> Using cache
 ---> 49f701ced622
Step 3/7 : COPY . .
 ---> 8bf02e99a224
Step 4/7 : RUN go build -o ./bin/goer ./cmd/goer
 ---> Running in 2c81a119585c
go: downloading github.com/sirupsen/logrus v1.4.2
go: downloading github.com/snowzach/rotatefilehook v0.0.0-20180327172521-2f64f265f58c
go: downloading github.com/gorilla/mux v1.8.0
go: downloading go.mongodb.org/mongo-driver v1.7.1
go: downloading github.com/gorilla/schema v1.2.0
go: downloading gopkg.in/natefinch/lumberjack.v2 v2.0.0
go: downloading golang.org/x/sys v0.0.0-20190531175056-4c3a928424d2
go: downloading github.com/youmark/pkcs8 v0.0.0-20181117223130-1be2e3e5546d
go: downloading github.com/go-stack/stack v1.8.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/golang/snappy v0.0.1
go: downloading github.com/klauspost/compress v1.9.5
go: downloading github.com/xdg-go/scram v1.0.2
go: downloading github.com/xdg-go/stringprep v1.0.2
go: downloading golang.org/x/crypto v0.0.0-20200709230013-948cd5f35899
go: downloading golang.org/x/sync v0.0.0-20200625203802-6e8e738ad208
go: downloading github.com/xdg-go/pbkdf2 v1.0.0
go: downloading golang.org/x/text v0.3.5
# github.com/eiffel-community/eiffel-goer/pkg/v1alpha1/handlers/events
pkg/v1alpha1/handlers/events/events.go:86:14: undefined: query.Parse
```
I suppose the problem is that deploy/goer/Dockerfile doesn't run `make gen` to generate the code from the .peg file. In fact, it doesn't use the makefile at all so it's also not building with govvv so the binary isn't version-stamped.

### Motivation
Docker image should be buildable.

### Exemplification
I want to build my own Docker image. How registry.nordix.org/eiffel/goer:5258723 was built is anyone's guess :-)

### Benefits
Obvious.

### Possible Drawbacks
None.

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -20,3 +20,21 @@ jobs:
 
       - name: Build and test
         run: make build check tidy check-dirty
+
+  build-docker-image:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: "recursive"
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v2
+        with:
+          # We need to specify the context explicitly so the action doesn't
+          # grab the source straight from the git. That's not a problem in
+          # itself but the .git directory (needed by govvv) won't be available.
+          context: .
+          file: deploy/goer/Dockerfile
+          pull: true

--- a/deploy/goer/Dockerfile
+++ b/deploy/goer/Dockerfile
@@ -1,7 +1,10 @@
 FROM golang:1.17.3-alpine AS build
+
+RUN apk update && apk add --no-cache git make
+
 WORKDIR /tmp/goer
 COPY . .
-RUN go build -o ./bin/goer ./cmd/goer
+RUN make build
 
 FROM alpine:3.13.5
 ENTRYPOINT ["/app/goer"]

